spec/requests/users_email_controller_spec.rb

```diff
@@ -9,11 +9,11 @@
   fab!(:moderator) { Fabricate(:moderator) }
 
   describe "#confirm-new-email" do
-    it 'redirects to login for signed out accounts' do
+    it 'does not redirect to login for signed out accounts, this route works fine as anon user' do
       get "/u/confirm-new-email/asdfasdf"
 
-      expect(response.status).to eq(302)
-      expect(response.redirect_url).to eq("http://test.localhost/login")
+      expect(response.status).not_to eq(302)
```

Comment:
We need to be careful when using `not_to` here. The `response.status` can return `500` and this test will still pass. Instead I think it is better to test for the exact `response.status` we expect.